### PR TITLE
Add support for Composer’s --ignore-platform-reqs option

### PR DESCRIPTION
**Pull request by drumm**
I had a need to use Composer's --ignore-platform-reqs option. This commit adds support for it.

### CODE REVIEW

File: templates/update.erb.rej [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff a/templates/update.erb b/templates/update.erb	(rejected hunks)
```
Comment (mattclegg) [inline templates/update.erb.rej]: Can you remove this file?

**Comment by drumm** [comment]
I rebased this to master, so it should now apply. The option can be used for composer update too, so the option should be added there.

**Comment by mattclegg** [comment]
Cool. Travis looks happier now.

Do you want to add it to composer update as part of this PR too?

I see you have rebased but that weird `templates/update.erb.rej`file still exists.

**Comment by drumm** [comment]
I see, I was misreading the files/comment. Pushed deleting that.

Thanks for the work upstream, I expect that could have helped out the tests.

**Pull request merged**